### DISCUSSION TITLE
ldns python confusion

### DISCUSSION THREAD

**Discussion by anandb-ripencc** [Homebrew/discussions#5484] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.9-33-g4bcdd81
ORIGIN: https://github.com/Homebrew/brew
HEAD: 4bcdd81657ddba8585b05818428317c8b06de0c6
Last commit: 4 hours ago
Core tap JSON: 11 Jul 07:11 UTC
Core cask tap JSON: 11 Jul 07:11 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_ENV_HINTS: set
HOMEBREW_NO_INSTALL_CLEANUP: set
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.3/bin/ruby
CPU: octa-core 64-bit icelake
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-x86_64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

The `ldns` formula on GitHub shows that it depends on python@3.12. However, on my system, I get this:
```
% brew uses --installed python@3.11
ldns
% brew list --versions ldns
ldns 1.8.3_1
```
I don't understand this. Apparently I have the latest version of `ldns`. Should I remove and reinstall it?

**Comment by SMillerDev** [comment]
If it was installed when it depended on 3.11 and you didn't reinstall it after it'll keep indicating 3.11. 

If you remove it and install it again it'll indicate that it depends on 3.12.

**Reply by anandb-ripencc** [reply, marked as answer]
Thanks @SMillerDev. That worked. But when the ldns dependencies change, shouldn't its revision be increased? I see that with many formulae.

**Reply by SMillerDev** [reply]
We only bump the revision when something changes for the user. This worked fine with 3.11 and with 3.12, people will generally get the change in the next package update since it's not urgent either.
